lib/Reader/readerir.cpp

```diff
@@ -3783,6 +3797,253 @@ IRNode *GenIR::genArrayElemAddress(IRNode *Array, IRNode *Index,
   return (IRNode *)Address;
 }
 
+bool GenIR::arrayGet(CORINFO_SIG_INFO *Sig, IRNode **RetVal) {
+  uint32_t Rank = 0;
+  CorInfoType ElemCorType = CORINFO_TYPE_UNDEF;
+  Type *ElementTy = nullptr;
+  const bool IsStore = false;
+  const bool IsLoadAddr = false;
+
+  if (!canExpandMDArrayRef(Sig, IsStore, IsLoadAddr, &Rank, &ElemCorType,
+                           &ElementTy)) {
+    return false;
+  }
+
+  IRNode *ElementAddress = mdArrayRefAddr(Rank, ElementTy);
+
+  // Load the value
+  assert(!ElementTy->isStructTy());
+  const bool IsVolatile = false;
+  LoadInst *LoadInst = makeLoadNonNull(ElementAddress, IsVolatile);
+  uint32_t Align = convertReaderAlignment(Reader_AlignNatural);
+  LoadInst->setAlignment(Align);
+  *RetVal = convertToStackType((IRNode *)LoadInst, ElemCorType);
+
+  return true;
+}
+
+bool GenIR::arraySet(CORINFO_SIG_INFO *Sig) {
+  uint32_t Rank = 0;
+  CorInfoType ElemCorType = CORINFO_TYPE_UNDEF;
+  Type *ElementTy = nullptr;
+  const bool IsStore = true;
+  const bool IsLoadAddr = false;
+
+  if (!canExpandMDArrayRef(Sig, IsStore, IsLoadAddr, &Rank, &ElemCorType,
+                           &ElementTy)) {
+    return false;
+  }
+
+  IRNode *Value = ReaderOperandStack->pop();
+
+  IRNode *ElementAddress = mdArrayRefAddr(Rank, ElementTy);
```

Comment:
Couldn't hurt to comment here that this will null-check the array and therefore we assume non-null at the store below.  Ditto for the mdArrayRefAddr call and load in arrayGet.